### ISSUE TITLE
Making a roundabout circular moves its speed bumps

### ISSUE THREAD

**Issue by 1ec5** [openstreetmap/iD#8265]
[This roundabout](https://www.openstreetmap.org/way/202355961) has speed bumps along it for some reason. Making it circular moves the speed bumps, which is inconsistent with how iD avoids moving connected ways when circularizing. This issue could conceivably occur with other features as well, such as a round building with entrance or loading dock nodes (but no footway or driveway connected to it yet), or an old-style traffic circle with `highway=traffic_signal` nodes at the stop lines.

**Comment by quincylvania** [comment]
Hmm I don't know how we could do this more intelligently. iD can't form a circle and leave more than three nodes in place. And forcing the circle to circumscribe two or three existing points might be unexpected. We don't know if the existing placement is accurate, perhaps the mapper wants the speed bumps to snap to the circle. Are you seeing drastic shifts?

> which is inconsistent with how iD avoids moving connected ways when circularizing

I'm not sure what you mean. iD will happily move junction nodes when circularizing.

**Comment by 1ec5** [comment]
> Hmm I don't know how we could do this more intelligently. iD can't form a circle and leave more than three nodes in place. And forcing the circle to circumscribe two or three existing points might be unexpected. We don't know if the existing placement is accurate, perhaps the mapper wants the speed bumps to snap to the circle. Are you seeing drastic shifts?

I see what you mean. In this case, the area was already mostly circular, but circularizing it caused the speed bumps to move around the circle. I would’ve expected the tagged vertex’s radial distance to change but not the angle, but that could well have been specific to the particular situation I was in.

> I'm not sure what you mean. iD will happily move junction nodes when circularizing.

It might not always be a big difference, but it seems like a connected way is constrained more than a tagged vertex. Compare the two configurations below, which are identical except that the configuration on the left has a connection to a roadway where the configuration on the right has a vertex tagged as a speed bump:

https://user-images.githubusercontent.com/1231218/103580727-8c88d680-4e8f-11eb-9bf4-4ae0a2a236cb.mov

In other words, attaching a temporary way to the speed bump is enough to keep it from shifting in the way that surprised me.
